linux/root_helpers_test.go

```diff
@@ -149,36 +153,111 @@ func randStr(n int) string {
 	return string(b)
 }
 
-func canUseLoop() (bool, string) {
-	loopCtrl := "/dev/loop-control"
-	euid := os.Geteuid()
-
-	if euid != 0 {
-		return false, fmt.Sprintf("uid is %d. must be 0.", euid)
+func isRoot() error {
```

Comment:
This seems like it should be a bool?

Reply:
well... by making it a error (maybe a bad function call) i could pass the result to t.Skipf() directly and have useful (string) reason in logs as to why it was being skipped.

Reply:
I generally feel like there are three possible states: 1. you are root, 2. you are not root, 3. you failed to detect whether you're not root. We used to use this same pattern (only a bool, only an error) in internal code, and it ended up masking bugs that we could have seen if we'd have used a (bool, error). Totally up to you, though.

I think you can still make a reasonable error message, just in a different place.

Reply:
these are just helpers ... for a test suite. only included in test code.

i'm confused though... essentially the return error provides a bool (err != nil) *and* a string. (string(err)).

what am i missing?

Reply:
Two states != three :). i.e. err == nil, err != nil, but what happens if you fail to detect root? That seems like a real problem with the test environment, and you don't want to just Skip() the test there.

Reply:
Anyway, feel free to ignore me. But we've been bitten by this type of thing in the past.

Reply:
Oh, I see your point though - geteuid() always succeeds. So ignore me. Might be nicer as a bool, but oh well.